### PR TITLE
Create repository item workflow

### PR DESCRIPTION
**Pull request by afolland**
Adds 'Create Repository Item' button to the default (HTML) view of self deposit webform submissions, per work item [in Airtable](https://airtable.com/apprqD9X6qd1EkZuf/tbliRGlWN4LaIIYhd/viwbx6KnAiZSrT0q9/receyj72ALdaxSa47/fldnikSB3WpPDMktR?copyLinkToCellOrRecordOrigin=gridView). 

To test, submit a test item via the Share Your Work form. The generated e-mail should have a link near the top that takes you to the webform submission. Taking that link should take you to a page that is an overview of the submission, with a 'Create Repository Item' button near the top of the page. Clicking this should create the repository item and add it to the self deposits collection.

### CODE REVIEW

File: web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        $child_node = $this->createNode($webform_submission, $values, $cfvalues['file_name'], $ftaxo_term, $copyright_term, $perm_term, $fmember_of);
+        $this->depositUtils->createMedia($cfvalues['media_type'], $cfvalues['field_name'], $cfkey, $child_node->id());
+      }
+    }
+    else {
+      $node = $this->createNode($webform_submission, $values, $values['item_title'], $taxo_term, $copyright_term, $perm_term, $member_of);
+      $this->depositUtils->createMedia($media_type, $field_name, $files[0], $node->id());
+    }
+
+    $webform_submission->setElementData('item_node', $node->id());
```
Comment (seth-shaw-asu) [inline web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php:232]: This doesn't appear to set a value. Do you need to do something like `$webform_submission->save();` for it to persist?

File: web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->createItem($this->webformSubmission);
+    $this->messenger()->addStatus('Successfuly created repository item.');
```
Comment (seth-shaw-asu) [inline web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php]: Can we get this to return a link to the new item node?

**Review by seth-shaw-asu — CHANGES_REQUESTED** [review]

File: web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php (hunk 1 above)
Comment (afolland) [reply web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php:232]: This whole function was copy-pasted from the existing webform handler. I'm not sure why it's there, that whole section could probably use a cleanup, but it also seems to work.

File: web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php (hunk 2 above)
Comment (afolland) [reply web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php]: Addressed in latest commit

**Review by seth-shaw-asu — APPROVED** [review]

**Comment by seth-shaw-asu** [comment]
👍
Go ahead and notify #repo-ops with testing instructions.

**Pull request merged**